common/faceting.js

```diff
@@ -641,16 +760,158 @@
                         return defer.promise;
                     };
 
+                    // Zoom the set into the middle 50% of the buckets
+                    scope.zoomInPlot = function () {
+                        try {
+                            // we don't want the user to keep zooming in infinitely to save memory space
+                            if (scope.histogramDataStack.length >= 20) { throw new Error("Maximum data stack size reached"); }
```

Comment:
What will happen if they exceed the maximum? Shouldn't we remove the `scope. histogramDataStack` in that case so they can continue using the zoom in? Or we want the zoom in to not be responsive?

Reply:
zoom will be unresponsive, a soft alert at top of page maybe?

Reply:
I added a `disableZoomIn()` function that disables the button that when clicked calls this function. Because I guard around the same condition, this case _should_ never be reached.

Reply:
Should I remove these error alerts instead?